### DISCUSSION TITLE
Getting 403 errors and a Captcha URL in response body of playlists queries

### DISCUSSION THREAD

**Discussion by duttonst** [tidal-music/discussions#86] [Report bug]
After successfully reading a 1,500 track playlist (and all of the many associated track queries I needed in order to retrieve artist & album details) I now seem to have been blocked from the endpoint for all playlist queries, e.g. `https://openapi.tidal.com/v2/playlists/me?include=items` no longer works.

I now only get 403 responses with the following body:
`{"url":"https://geo.captcha-delivery.com/captcha/?(lots of parameters here)"}`

A couple of points:
1. This doesn't seem like a useful response for a client to receive from a RESTful endpoint
2. How can I get unblocked?! (Edit: I seemed to get unblocked after a period of time and have set up an alternative App Id in case it happens again).

Using 429 responses would be a more conventional way to ask a client to back-off. I did actually receive some 429s and did back-off when requested, but I still ended-up getting blocked.

**Comment by ps0uth** [comment]
I'm getting the same thing when trying to browse the deveoper documentation  or any page at under developer.tidal.com. If I had to guess there is either something going haywire with their security solution, or misconfiguration in their cloudfront caching policies, or none of the above. ONCE it delivered me a captcha to solve which bought me a few minutes of looking through the api docs, then it showed up again without any possibility of getting through it. :disappointed: 

![image](https://github.com/user-attachments/assets/81e7e973-f982-4448-8aef-4e7c690909d0)

**Reply by duttonst** [reply]
Presumably an API client app could easily trigger 1 or 3 on that list if things are misconfigured. I buy your theory.

I'm finding that everything on the API seems to be limited to page sizes of 20, meaning that even reading a playlist requires so many queries it probably looks like a DDoS attack to the security gateway.

**Comment by stefanmoro** [comment]
Hey @duttonst, @philipsoutham, thanks for reaching out.

It seems our bot protection got a bit to eager here.

We're still considering TIDAL API to be in "beta" as we work on expanding functionality and tuning other aspect of it. We'll look into this specific situation next week. Thank you for your patience.

**Reply by duttonst** [reply]
Many thanks!
